bionic/flow.py

```diff
@@ -1231,9 +1232,34 @@ def get(self, name, collection=None, fmt=None, mode=object):
             return set(values)
         elif collection is pd.Series or collection == "series":
             if len(result_group.key_space) > 0:
+                case_keys = [result.query.case_key for result in result_group]
+
+                resultgroups_by_entity_name = {
+                    entity_name: self._deriver.derive(
+                        entity_dnode_from_descriptor(entity_name)
+                    )
+                    for entity_name in result_group.key_space
+                }
+                values_by_entity_name_by_case_key = defaultdict(lambda: defaultdict())
```

Comment:
Why is the inner dict a `defaultdict` instead of a regular `dict`?

But also, rather than looping through every combination of original and ancestor case key, it might be cleaner to organize the ancestor values according to the *ancestor* case key. Then you can look them up like this:

```
ancestor_values_by_case_key_by_name[ancestor_name][orig_case_key.project(ancestor_key_space)]
```

Reply:
Thanks, this was very helpful!